jax/_src/numpy/lax_numpy.py

```diff
@@ -1597,16 +1598,24 @@ def stack(arrays, axis: int = 0, out=None):
 def tile(A, reps):
   _stackable(A) or _check_arraylike("tile", A)
   try:
-    iter(reps)
+    reps = iter(reps)
   except TypeError:
     reps = (reps,)
   reps = tuple(operator.index(rep) if core.is_constant_dim(rep) else rep
                for rep in reps)
   A_shape = (1,) * (len(reps) - ndim(A)) + shape(A)
   reps = (1,) * (len(A_shape) - len(reps)) + reps
-  result = broadcast_to(reshape(A, [j for i in A_shape for j in [1, i]]),
-                        [k for pair in zip(reps, A_shape) for k in pair])
-  return reshape(result, tuple(np.multiply(A_shape, reps)))
+  merged_rs = [(r * s,) if type(r) is type(s) is int and (r == 1 or s == 1)
```

Comment:
Is the comparison `type(r) is int` meant to address the case when `r` is a non-constant dimension? In the case, it is better to use `core.is_constant_dim`, as above. Also, `r == 1 or s == 1` should be written as `core.symbolic_equal_dim(r, 1) or core.symbolic_equal_dim(s, 1)`

Reply:
Thanks! I will have a look.

Reply:
I use `core.symbolic_equal_dim(r, 1) or core.symbolic_equal_dim(s, 1)` instead of `type(r) is type(s) is int and (r == 1 or s == 1)`. Thanks!